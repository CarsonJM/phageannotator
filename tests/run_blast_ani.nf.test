nextflow_pipeline {

    name "Test pipeline: NFCORE_PHAGEANNOTATOR"
    script "main.nf"
    tag "PIPELINE"


    test("run_blast_ani = true") {
        when {
            params {
                input           = "$projectDir/assets/test-datasets/samplesheets/two_samples_no_reads.csv"
                outdir          = "$outputDir"
                run_blast_ani   = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${params.outdir}/ANICalculation/checkvanicluster/aanicalc")
                    ).match() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() }
            )
        }
    }
}
