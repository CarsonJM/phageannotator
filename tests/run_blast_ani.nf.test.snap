{
    "run_blast_ani = true": {
        "content": [
            [
                "all_samples_ani.tsv:md5,d2781470e7b8248cf7a1c76b0744e48b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T11:57:42.026692073"
    }
}